lib/hydra/derivatives/runners/active_encode_derivatives.rb

```diff
@@ -1,22 +1,21 @@
 module Hydra::Derivatives
   class ActiveEncodeDerivatives < Runner
+    # TODO: object_or_filename - I'm currently passing a String
+    # filename during my testing, but we'll probably need to
+    # change this so that we can pass in the FileSet or File
+    # object that the source file is attached to.
+    #
     # @param [String, ActiveFedora::Base] object_or_filename path to the source file, or an object
     # @param [Hash] options options to pass to the encoder
-    # @options options [Array] :outputs a list of desired outputs, each entry is a hash that has :label (optional), :format and :url
+    # @options options [Array] :outputs a list of desired outputs
     def self.create(object_or_filename, options)
-      source_file(object_or_filename, options) do |f|
-        instructions = transform_directives(options)
-        processor_class.new(f,
-                            instructions.merge(source_file_service: source_file_service),
-                            output_file_service: output_file_service).process
+      file_name = object_or_filename
+      transform_directives(options.delete(:outputs)).each do |instructions|
```

Comment:
I'm not quite understanding the reason to switch from all H-D directives being run by a single processor instance to each H-D directive getting its own processor.  Is there a case where one processor can't handle all of the directives at once?  Another way to ask this is is there any case where a call to ActiveEncodeDerivatives.create should spawn multiple ActiveEncode::Base.create calls?